### PR TITLE
Fix #1907: Support svg image in LaTeX output.

### LINKED ISSUE #1907: no SVG support in PDFs
Coming from https://github.com/rtfd/readthedocs.org/issues/905 here... it looks like Sphinx doesn't support SVGs properly. Or, more accurately, the LaTeX file it generates asumes that `\includegraphics` can deal with SVG files: it can't.

In LaTeX, SVG images are special: they need to be first converted into another format using Inkscape, and the SVG package. More information about this is available in [this TeX stackexchange question](http://tex.stackexchange.com/questions/122871/include-svg-images-with-the-svg-package), but basically, if a `.svg` file extension is detected, sphinx should generate the following, in the preambule:

```
\usepackage{svg}
```

and instead of `\includegraphics`, use:

```
\includesvg{file}
```

Notice how the `.svg` file extension should _not_ be provided. It may be possible to hack up something to replace the `\includegraphics{}` directive so it calls `\includesvg` when there's a `.svg` extension, but i would recommend against that confusing tactic and generate the right directive straight from Sphinx.

Of course, [Inkscape](http://inkscape.org) needs to be available (in the path, or the path can be specified in the preambule, with `\usepackage[inkscape={/opt/bin/inkscap‌​e -z -C}]{svg}`) and `--shell-escape` must be passed on the commandline. It may be interesting to allow inkscape parameters to be overridable from the `conf.py` of course.

A workaround is to generate a PNG file by hand on the side, then include the right file depending on the ouput format:

```
.. only:: latex

  .. image:: file.png

.. only:: html

  .. image:: file.svg
```

Then, to generate `file.png`, one could add this to the makefile:

```
%.pdf: %.svg
    inkscape --without-gui --export-area-drawing --file=$< --export-pdf=$@
```

You could also add targets to the makefile to make sure those images are built automatically:

```
SOURCEDIR     = source
#IMAGEDIRS can be a list of directories that contain SVG files and are relative to the SOURCEDIR
IMAGEDIRS      = _images _images2 ...

PDFs := $(foreach dir, $(IMAGEDIRS), $(patsubst %.svg,%.pdf,$(wildcard $(SOURCEDIR)/$(dir)/*.svg)))

images: $(PDFs)

# make sure PDFs are cleaned
clean:
    -rm -rf $(BUILDDIR)/*
    -rm $(PDFs)

latex: images
    # [...]
latexpdf: images
    # [...]
```

(Inspired in part by [this tutorial](https://sites.google.com/site/nickfolse/home/sphinx-latexpdf-output-with-svg-images).)

But that's kind of a horrible hack that shouldn't be necessary.

### PR DESCRIPTION
**Pull request by xuhdev**
However this relies on an existing installation of inkscape. The tests of svg should be ignored if no inkscape installation is found.

### CODE REVIEW

**Comment by xuhdev** [comment]
This PR should solve #1907

File: sphinx/writers/latex.py [hunk 1]
```diff
@@ -1182,7 +1184,11 @@ def visit_image(self, node):
         options = ''
         if include_graphics_options:
             options = '[%s]' % ','.join(include_graphics_options)
-        self.body.append('\\includegraphics%s{%s}' % (options, uri))
+        if node['uri'].lower().endswith('svg') or \
```
Comment (tk0miya) [inline sphinx/writers/latex.py:1187]: I think `endswith(('.svg', '.svgz'))` is better.

**Comment by tk0miya** [comment]
It seems this is a new feature. The `master` branch is more appropriate to apply this PR.
Could you resend it again?

**Comment by tk0miya** [comment]
In addition, it seems enabling `--shell-escape` is insecure operation.
It should be better that the user specify any settings consciously (`conf.py` or `texmf.cnf`).
What do you think about the option?

File: sphinx/writers/latex.py (hunk 1 above)
Comment (xuhdev) [inline sphinx/writers/latex.py:1187]: Thanks, I agree with you.

**Comment by xuhdev** [comment]
@tk0miya Yes, I can add an option in conf.py. How about `latex_shell_escaped = True | False(default)`? I would also add an error message when svg files are included but this option is not turned on. And yes, I will resend the PR for master.

**Comment by xuhdev** [comment]
I think I have a better way to implement this without touching the `--shell-escaped` setting. We can convert the image format (`pdf` to `svg` for html output and `svg` to `pdf` for LaTeX output) using inkscape before generate the documents. I can reimplement these on master.

**Comment by tk0miya** [comment]
New idea looks good to me!

And, I think the feature should be implemented as sphinx extension (cf. `sphinx.ext.latexsvg` ).
This feature depends on `inkscape`. If we compose it as extension, it keeps sphinx-core simple and less dependencies.

@birkenfeld @shimizukawa What do you think about this?

**Comment by xuhdev** [comment]
@tk0miya It is more than LaTeX and svg --- it can also convert pdf/eps to svg for html output. Maybe we can call it `compatible_images`?

**Comment by tk0miya** [comment]
Oh, sorry. I did not noticed your comments...

It might the users can not guess what it does if it is named as `compatible_images`.
I think more direct words are better like 'convert'.

The naming of the extensions is very important, I think. but, at the same time, it is very difficult to get an agreements about it.
So, in this time, please name it as you like.

**Comment by xuhdev** [comment]
Thanks, I'll name it as auto_image_converter . I'll make another PR :)

**Comment by mgeier** [comment]
In case you're not aware of it, the `nbconvert` package has a converter using Inkscape: https://github.com/jupyter/nbconvert/blob/master/nbconvert/preprocessors/svg2pdf.py

About the name: Since it depends on Inkscape, probably it should have `inkscape` in its name?
Something like `sphinx.ext.inkscapeconverter` perhaps?

**Pull request closed without merging**